### DISCUSSION TITLE
Is it possible to let jdtls work after i save a file?

### DISCUSSION THREAD

**Discussion by SpicyChickenFLY** [mfussenegger/nvim-jdtls#482] [Q&A] [answered]
I was struggled that JDTLS always updates the workspace immediately after I input something, which makes me wait for a long time before completion can be used again. Is it possible make JDTLS stop until I save the file?

**Comment by mfussenegger** [comment, marked as answer]
You can disable autobuild:

```lua
config.settings = {
  java = {
    autobuild = { enabled = false },
  }
}
```

`config` is referring to the `config` from https://github.com/mfussenegger/nvim-jdtls#configuration-verbose



If you're using nvim-jdtls with nvim-dap you might still want to trigger a blocking build when you save a document, otherwise you'll debug a stale/outdated version of your code.

E.g. you could add something like:

```lua
config.on_attach = function(client, bufnr)
  api.nvim_create_autocmd("BufWritePost", {
    buffer = bufnr,
    callback = function()
      client.request_sync("java/buildWorkspace", false, 5000, bufnr)
    end,
  })
end
```

**Reply by oysandvik94** [reply]
Maybe I am misunderstanding how this works, but when I set this setting, and i.e. change my file by deleting a random character, I immediately get updated diagnostic information, it does not happen on save. Is the autobuild referring to something else?
